apps/web/components/auth-card.tsx

```diff
@@ -0,0 +1,47 @@
+import Link from "next/link";
+
+import {
+  Card,
+  CardContent,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "@repo/ui/components/card";
+import { Button } from "@repo/ui/components/button";
+import { AuthForm } from "./auth-form";
+
+type Props = {
+  isSignUp?: boolean;
```

Comment:
Having this boolean could lead to loosing flexibility and maintainability of your code. Think about `variant` prop which might be more flexible.

e.g.  
```suggestion
  variant: "sign-up" | "sign-in";
```

Reply:
Then you can create a function returning required values based on that variant

Reply:
Here is the whole snippet of the code:

```tsx
type AuthCardVariant = "sign-in" | "sign-up";

type CardContent = {
  title: string;
  linkText: string;
  link: string;
};

const SIGN_IN_CARD_CONTENT: CardContent = {
  title: "Log in",
  linkText: "sign up here",
  link: "/sign-up",
};

const SIGN_UP_CARD_CONTENT: CardContent = {
  title: "Sign up",
  linkText: "login in here",
  link: "/log-in",
};

const AUTH_CARD_CONTENT: Record<AuthCardVariant, CardContent> = {
  "sign-in": SIGN_IN_CARD_CONTENT,
  "sign-up": SIGN_UP_CARD_CONTENT,
};

function getCardContent(variant: AuthCardVariant): CardContent {
  return AUTH_CARD_CONTENT[variant] || SIGN_IN_CARD_CONTENT;
}

// ...
function AuthCard({ variant }: Props) {
  const { link, linkText, title } = getCardContent(variant);
```

Reply:
The same result, but it looks much clearer and cleaner from the code perspective